pkg/test/utils.go

```diff
@@ -174,19 +174,19 @@ func CompareGoldenObject(t *testing.T, p string, got []byte) {
 		return
 	}
 
+	t.Errorf("unexpected diff in %s: %s", p, diff)
+
 	if writeGoldenOutput {
 		// Write the output to the golden file
 		if err := os.WriteFile(p, []byte(got), 0644); err != nil {
 			t.Fatalf("failed to write golden output %s: %v", p, err)
 		}
-		t.Errorf("wrote updated golden output to %s", p)
-	} else {
-		t.Errorf("unexpected diff in %s: %s", p, diff)
+		t.Logf("wrote updated golden output to %s", p)
 	}
 }
 
 func postWriteNormalization(wantMap map[string]interface{}) map[string]interface{} {
-	if os.Getenv("KCC_USE_DIRECT_RECONCILERS") != "" {
+	if os.Getenv("XXX_KCC_USE_DIRECT_RECONCILERS") != "" {
```

Comment:
is this intended?

Reply:
Yes, it's a quick hack while this is WIP!

Reply:
That said it was confusing, so I just deleted it instead of hacking-it-out